examples/qml_features/rust/src/custom_base_class.rs

```diff
@@ -158,6 +158,15 @@ pub mod qobject {
     }
     // ANCHOR_END: book_inherit_can_fetch_more_signature
 
+    /// Roles for the QAbstractListModel
+    #[repr(i32)]
+    enum Roles {
```

Comment:
As this is only used on the Rust side and not the C++/QML side, do we actually need to declare this as a shared enum ?

Or is this good to demo that they could cast the `Roles::Id` in C++ to an `int` and that would then work with the `data` method or if we could later expose that as a QML enum it would work ?